### PR TITLE
[WIP] make the appearence of voltage values in sld consistent if loadflow i…

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**
…nvalid

Signed-off-by: walid-sahnoun <walid.sahnoun@rte-france.com>

### CODE REVIEW

**Comment by etiennehomer** [comment]
You can do something more simple : you can retrieve the element <text class="sld-label" id="NODE_ACHIEP4_0_v" x="60.0" y="600.0">94.7 kV</text> with the methods getElementByClassName, getParent, getChildren...
And then you change its text context.

**Review by geofjamg — CHANGES_REQUESTED** [review]
Definitely not the right way to do it

File: src/components/diagrams/singleLineDiagram/single-line-diagram.js [hunk 1]
```diff
@@ -370,37 +370,33 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
...
+                    for (var i = 0; i < element.length; i++) {
+                        if (element[i].id.endsWith('_circle')) {
+                            nodeIdKV = element[i].id.replace('circle', 'v');
+                            nodeIdAngle = element[i].id.replace(
+                                'circle',
+                                'angle'
+                            );
+                        }
+                    }
+                    if (nodeIdKV !== null && nodeIdKV !== undefined) {
```
Comment (etiennehomer) [inline src/components/diagrams/singleLineDiagram/single-line-diagram.js:389]: if(nodeIdKV)

File: src/components/diagrams/singleLineDiagram/single-line-diagram.js [hunk 2]
```diff
@@ -370,37 +370,33 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
...
+                        let oldValue =
+                            document.getElementById(nodeIdKV).textContent;
+                        svgToUpdate = svgToUpdate.replace(oldValue, '— kV');
+                    }
+                    if (nodeIdAngle !== null && nodeIdAngle !== undefined) {
```
Comment (etiennehomer) [inline src/components/diagrams/singleLineDiagram/single-line-diagram.js:394]: if(nodeIdAngle)

File: src/components/diagrams/singleLineDiagram/single-line-diagram.js [hunk 3]
```diff
@@ -370,37 +370,33 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
...
+            if (svg['svg'] !== null) {
+                var svgToUpdate = svg['svg'];
+                let element =
+                    document?.getElementsByClassName('sld-node-infos');
+                if (element != null) {
+                    let nodeIdKV;
+                    let nodeIdAngle;
+                    for (var i = 0; i < element.length; i++) {
+                        if (element[i].id.endsWith('_circle')) {
+                            nodeIdKV = element[i].id.replace('circle', 'v');
```
Comment (etiennehomer) [inline src/components/diagrams/singleLineDiagram/single-line-diagram.js:382]: You can build nodeIdKV and nodeIdAngle with : 
'NODE_' + sldId + '_0_v'

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/diagrams/singleLineDiagram/single-line-diagram.js [hunk 4]
```diff
@@ -368,6 +368,40 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
         sldId,
     ]);
 
+    useEffect(() => {
+        if (loadFlowStatus !== RunningStatus.SUCCEED) {
+            if (svg['svg'] !== null) {
+                var svgToUpdate = svg['svg'];
+                let element =
```
Comment (etiennehomer) [inline src/components/diagrams/singleLineDiagram/single-line-diagram.js:375]: rename to nodeInfos

File: src/components/diagrams/singleLineDiagram/single-line-diagram.js [hunk 5]
```diff
@@ -368,6 +368,40 @@ const SizedSingleLineDiagram = forwardRef((props, ref) => {
...
+                    document?.getElementsByClassName('sld-node-infos');
+                if (element != null) {
+                    let nodeIdKV;
+                    let nodeIdAngle;
+                    for (var i = 0; i < element.length; i++) {
+                        if (element[i].id.endsWith('_circle')) {
```
Comment (etiennehomer) [inline src/components/diagrams/singleLineDiagram/single-line-diagram.js:381]: you have to replace the text for each

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Review by etiennehomer — CHANGES_REQUESTED** [review]
Remove package-lock.json changes

**Pull request closed without merging**